stream-chat-android-client/src/main/java/io/getstream/chat/android/client/helpers/QueryChannelsPostponeHelper.kt

```diff
@@ -46,12 +46,10 @@ internal class QueryChannelsPostponeHelper(
         check(attemptCount > 0) {
             "Failed to perform job. Waiting for set user completion was too long. Limit of attempts was reached."
         }
-        return when (clientStateService.state) {
-            is ClientState.Idle -> error("User must be set before querying channels")
-            is ClientState.User.Authorized,
-            is ClientState.Anonymous.Authorized -> job()
-            is ClientState.User.Pending,
-            is ClientState.Anonymous.Pending -> {
+        return when (socketStateService.state) {
+            is SocketState.Idle -> error("Socket connection must be established before querying channels")
```

Comment:
What about if we allow user to use our ChatClient before the `SocketState` is initialized?
We can follow the same "delay" we have for Pending and Disconnected, and after X attempts, it will fail if they didn't connected